neumai/neumai/SinkConnectors/LanceDBSink.py

```diff
@@ -0,0 +1,158 @@
+from neumai.Shared.NeumSinkInfo import NeumSinkInfo
+from neumai.Shared.NeumVector  import NeumVector
+from neumai.Shared.NeumSearch import NeumSearchResult
+from neumai.Shared.Exceptions import(
+    LanceDBInsertionException,
+    LanceDBIndexInfoException,
+    LanceDBQueryException
+)
+from neumai.SinkConnectors.SinkConnector import SinkConnector
+from typing import List
+from pydantic import Field
+
+import lancedb
+from lancedb import DBConnection
+
+
+class LanceDBSink(SinkConnector):
+
+    """
+    LanceDB sink
+
+    A sink connector for LanceDB, designed to facilitate data output into a 
+    LanceDB storage system. For details about LanceDB, refer to 
+    https://github.com/lancedb/lancedb.
+
+
+    Attributes:
+    -----------
+    uri: str
+        URI for LanceDB database.
+    api_key: str
+        If presented, connect to LanceDB cloud. 
+        Otherwise, connect to a database on file system or cloud storage.
+    region: str
+        Region for use of LanceDB cloud.
+    table_name: str
+        Name of LanceDB table to use
+    create_index: bool
+        LanceDB offers flat search as well as ANN search. If set to True, 
+        a vector index will be created for searching instead of a brute-force
+        knn search.
+
+
+    Example usage:
+        ldb = LanceDBSink(uri="data/test_ldb_sink", table_name="demo_ldb_table")
+        ldb.store(neum_vectors)
+        ldb.search(query)
+    """
+
+    uri: str = Field(..., description="URI for LanceDB database")
+    api_key: str = Field(default=None, description="API key for LanceDB cloud")
+    region: str = Field(default=None, description="Region for use of LanceDB cloud")
```

Comment:
Are these optional fields? If so, type should be Optional[str]

Reply:
Yes they are optional, made the changes.